test/framework/src/main/java/org/elasticsearch/AnalysisFactoryTestCase.java

```diff
@@ -0,0 +1,326 @@
+        put("hyphenatedwords",           Void.class);
+        // repeats anything marked as keyword
+        put("keywordrepeat",             Void.class);
+        // like limittokencount, but by offset
+        put("limittokenoffset",          Void.class);
+        // like limittokencount, but by position
+        put("limittokenposition",        Void.class);
+        // ???
+        put("numericpayload",            Void.class);
+        // removes duplicates at the same position (this should be used by the existing factory)
+        put("removeduplicates",          Void.class);
+        // ???
+        put("tokenoffsetpayload",        Void.class);
+        // puts the type into the payload
+        put("typeaspayload",             Void.class);
+        // fingerprint
+        put("fingerprint",               Void.class);
+        // for tee-sinks
+        put("daterecognizer",            Void.class);
+    }});
+    
+    static final Map<String,Class<?>> KNOWN_CHARFILTERS = Collections.unmodifiableMap(new HashMap<String,Class<?>>() {{        
+        // exposed in ES
+        put("htmlstrip",      HtmlStripCharFilterFactory.class);
+        put("mapping",        MappingCharFilterFactory.class);
+        put("patternreplace", PatternReplaceCharFilterFactory.class);
+                
+        // TODO: these charfilters are not yet exposed: useful?
+        // handling of zwnj for persian
+        put("persian",        Void.class);
+    }});
+
+    protected Map<String, Class<?>> getTokenizers() {
+        return KNOWN_TOKENIZERS;
+    }
+
+    protected Map<String, Class<?>> getTokenFilters() {
+        return KNOWN_TOKENFILTERS;
+    }
+
+    protected Map<String, Class<?>> getCharFilters() {
+        return KNOWN_CHARFILTERS;
+    }
+
+    public void testTokenizers() {
+        Set<String> missing = new TreeSet<String>(org.apache.lucene.analysis.util.TokenizerFactory.availableTokenizers());
+        missing.removeAll(getTokenizers().keySet());
+        assertTrue("new tokenizers found, please update KNOWN_TOKENIZERS: " + missing.toString(), missing.isEmpty());
+    }
+
+    public void testCharFilters() {
+        Set<String> missing = new TreeSet<String>(org.apache.lucene.analysis.util.CharFilterFactory.availableCharFilters());
+        missing.removeAll(getCharFilters().keySet());
+        assertTrue("new charfilters found, please update KNOWN_CHARFILTERS: " + missing.toString(), missing.isEmpty());
+    }
+
+    public void testTokenFilters() {
+        Set<String> missing = new TreeSet<String>(org.apache.lucene.analysis.util.TokenFilterFactory.availableTokenFilters());
+        missing.removeAll(getTokenFilters().keySet());
+        assertTrue("new tokenfilters found, please update KNOWN_TOKENFILTERS: " + missing.toString(), missing.isEmpty());
+    }
+
+    public void testMultiTermAware() {
+        Collection<Class<?>> expected = new HashSet<>();
+        for (Map.Entry<String, Class<?>> entry : getTokenizers().entrySet()) {
```

Comment:
hmm actually there is not much that I can extract to a method since it is calling a different static method each time?